### PR TITLE
man.vim: bug fixes

### LINKED ISSUE #5628: man.vim: no syntax highlighting (+ error) for local man files
- `nvim --version`: 

NVIM v0.2.0-97-gacbce46
Build type: RelWithDebInfo

- Operating system/version: Ubuntu 14.04
- Terminal name/version: urxvt
- `$TERM`: rxvt-unicode

### Actual behaviour

Local man files provoke an error and aren't syntax highlighted.

### Expected behaviour

I want to see local man files the same way as man files that are in `$MANPATH`.

### Steps to reproduce using `nvim -u NORC`

```
$ export MANPAGER=nvim '+set ft=man' -
$ man -l zstd.1
Error detected while processing function man#extract_sect_and_name_ref:                                                                                                                                                                                        
line    8:                                                                                                                                                                                                                                                     
E605: Exception not caught: manpage reference cannot contain only parentheses                                                                                                                                                                                  
Error detected while processing /usr/home/aktau/neovim/share/nvim/runtime/ftplugin/man.vim:                                                                                                                                                       
line   23:                                                                                                                                                                                                                                                     
E171: Missing :endif                                                                                                                                                                                                                                           
Error detected while processing function <SNR>81_LoadFTPlugin:                                                                                                                                                                                                 
line   17:                                                                                                                                                                                                                                                     
E170: Missing :endfor                  
```

### PR DESCRIPTION
**Pull request by nhooyr**
Fixes #5628 and #5574

### CODE REVIEW

**Comment by nhooyr** [comment]
O, I got a better idea to fix this.

**Comment by nhooyr** [comment]
I got a better way to fix this.

File: runtime/autoload/man.vim [hunk 1]
```diff
@@ -50,7 +50,7 @@ function! man#open_page(count, count1, mods, ...) abort
     return
   endtry
   call s:push_tag()
-  let bufname = 'man://'.name.(empty(sect)?'':'('.sect.')')
+  let bufname = fnameescape('man://'.name.(empty(sect)?'':'('.sect.')'))
```
Comment (justinmk) [inline runtime/autoload/man.vim]: better to call fnameescape() directly where it actually needs to be escaped. Or else rename this variable to `bufname_esc`

Comment (nhooyr) [inline runtime/autoload/man.vim]: fixed

**Comment by justinmk** [comment]
Thanks @nhooyr

**Pull request merged**